### PR TITLE
feat(dataarts): add resource dataarts studio data recognition rule

### PR DESCRIPTION
**Pull request by zhangting951007**
**What this PR does / why we need it**:
add a new resource dataarts studio data recognition rule
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
+ You need to prepare workspace_id, secrecy_level_id and category_id before use.
+ When the rule is not exist, the detail API response body is: {"error_code": "DLS.4106","error_msg": "Rule is not exist."}
image

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/dataarts" TESTARGS="-run TestAccResourceRecognitionRule_custom"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run TestAccResourceRecognitionRule_custom -timeout 360m -parallel 4
=== RUN   TestAccResourceRecognitionRule_custom
=== PAUSE TestAccResourceRecognitionRule_custom
=== CONT  TestAccResourceRecognitionRule_custom
--- PASS: TestAccResourceRecognitionRule_custom (31.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  31.702s
make testacc TEST="./huaweicloud/services/acceptance/dataarts" TESTARGS="-run TestAccResourceRecognitionRule_builtin"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run TestAccResourceRecognitionRule_builtin -timeout 360m -parallel 4
=== RUN   TestAccResourceRecognitionRule_builtin
=== PAUSE TestAccResourceRecognitionRule_builtin
=== CONT  TestAccResourceRecognitionRule_builtin
--- PASS: TestAccResourceRecognitionRule_builtin (21.74s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  21.780s
```

### CODE REVIEW

File: docs/resources/dataarts_studio_data_recognition_rule.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+subcategory: "DataArts Studio"
+---
+
+# huaweicloud_dataarts_studio_rule
+
+Manages DataArts Studio rule resource within HuaweiCloud.
```
Comment (deer-hang) [inline docs/resources/dataarts_studio_data_recognition_rule.md]:
```suggestion
Manages a DataArts Studio data recognition rule resource within HuaweiCloud.
```

File: docs/resources/dataarts_studio_data_recognition_rule.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+* `enable` - Specifies whether the current data recognition rule is available.
+
+
+## Import
+
+DataArts Studio rule can be imported using `<workspace_id>/<id>`, e.g.
+
+```sh
+terraform import huaweicloud_dataarts_studio_rule.test b606cd4a47b645108a122857204b360f/82dde23fb5d94838ac34f27a9e390bd0
```
Comment (deer-hang) [inline docs/resources/dataarts_studio_data_recognition_rule.md]:
```suggestion
terraform import huaweicloud_dataarts_studio_data_recognition_rule.test <workspace_id>/<id>
```

File: docs/resources/dataarts_studio_data_recognition_rule.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
+---
+subcategory: "DataArts Studio"
+---
+
+# huaweicloud_dataarts_studio_rule
```
Comment (deer-hang) [inline docs/resources/dataarts_studio_data_recognition_rule.md]:
```suggestion
# huaweicloud_dataarts_studio_data_recognition_rule
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_recognition_rule.go [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+	deleteRulePath = strings.ReplaceAll(deleteRulePath, "{id}", d.Id())
+
+	deleteRuleOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	_, err = deleteRuleClient.Request("DELETE", deleteRulePath, &deleteRuleOpt)
+	if err != nil {
+		return diag.FromErr(err)
```
Comment (deer-hang) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_recognition_rule.go]:
```suggestion
		return diag.Errorf("error deleting DataArts Studio data recognition rule: %s", err)
```

**Comment by zhangting951007** [comment]
```
make testacc TEST="./huaweicloud/services/acceptance/dataarts" TESTARGS="-run TestAccResourceRecognitionRule_custom"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run TestAccResourceRecognitionRule_custom -timeout 360m -parallel 4
=== RUN   TestAccResourceRecognitionRule_custom
=== PAUSE TestAccResourceRecognitionRule_custom
=== CONT  TestAccResourceRecognitionRule_custom
--- PASS: TestAccResourceRecognitionRule_custom (32.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  32.420s
```
```
make testacc TEST="./huaweicloud/services/acceptance/dataarts" TESTARGS="-run TestAccResourceRecognitionRule_builtin"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run TestAccResourceRecognitionRule_builtin -timeout 360m -parallel 4
=== RUN   TestAccResourceRecognitionRule_builtin
=== PAUSE TestAccResourceRecognitionRule_builtin
=== CONT  TestAccResourceRecognitionRule_builtin
--- PASS: TestAccResourceRecognitionRule_builtin (21.36s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  21.424s
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_recognition_rule.go [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceStudioRule() *schema.Resource {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_recognition_rule.go:25]: please add the used APIs at the begin of this resource.

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_recognition_rule.go [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+
+	return resourceRuleRead(ctx, d, meta)
+}
+
+func resourceRuleRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	workspaceID := d.Get("workspace_id").(string)
+
+	//nolint:misspell
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_recognition_rule.go]: we can delete this line

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_recognition_rule.go [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+		d.Set("content_expression", utils.PathSearch("content_expression", getRuleRespBody, nil)),
+		d.Set("column_expression", utils.PathSearch("column_expression", getRuleRespBody, nil)),
+		d.Set("comment_expression", utils.PathSearch("commit_expression", getRuleRespBody, nil)),
+		d.Set("builtin_rule_id", utils.PathSearch("builtin_rule_id", getRuleRespBody, nil)),
+		d.Set("category_id", utils.PathSearch("category_id", getRuleRespBody, nil)),
+		d.Set("description", utils.PathSearch("description", getRuleRespBody, nil)),
+		d.Set("secrecy_level", utils.PathSearch("secrecy_level", getRuleRespBody, nil)),
+		d.Set("secrecy_level_num", utils.PathSearch("secrecy_level_num", getRuleRespBody, nil)),
+		d.Set("enable", utils.PathSearch("enable", getRuleRespBody, nil)),
+		d.Set("created_at", utils.FormatTimeStampUTC(int64(utils.PathSearch("created_at", getRuleRespBody, nil).(float64))/1000)),
```
Comment (ShiChangkuo) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_recognition_rule.go]: suggest to use `FormatTimeStampRFC3339`

**Pull request merged**